Mathlib/NumberTheory/EllipticDivisibilitySequence.lean

```diff
@@ -58,36 +84,248 @@ polynomials of elliptic curves, omitting a factor of the bivariate `2`-division
 
 ## References
 
-M Ward, *Memoir on Elliptic Divisibility Sequences*
+* K Stange, *Elliptic Nets and Elliptic Curves*
+* M Ward, *Memoir on Elliptic Divisibility Sequences*
 
 ## Tags
 
-elliptic, divisibility, sequence
+elliptic net, elliptic divisibility sequence
 -/
 
 universe u v
 
-variable {R : Type u} [CommRing R]
+variable {R : Type u} {S : Type v} [CommRing R] [CommRing S] (W : ℤ → R) (f : R →+* S)
+
+namespace IsEllipticNet
+
+/-- The elliptic atom `Wₐ(p, q)` that defines an elliptic net. Note that this is defined in terms of
+truncated integer division, and hence should only be used when `p` and `q` have the same parity. -/
+def atom (p q : ℤ) : R :=
+  W ((p + q).tdiv 2) * W ((p - q).tdiv 2)
```

Comment:
Is there a reason to prefer `tdiv` over simply `/` (which seems to be the same as `Int.ediv`)?